html/rendering/widgets/baseline-alignment-and-overflow.tentative.html

```diff
@@ -0,0 +1,181 @@
+  <tr class="ref-file"><td><span>ref file</span> <b><button>file</button></b>
+  <tr class="ref-image-alt"><td><span>ref image</span> <b>alt</b></b>
+  <tr class="ref-image-src"><td><span>ref image</span> <img class=set-src>
+</table>
+<h2>tests</h2>
+<table class="test" id="test-table"></table>
+<script>
+  "use strict";
+
+  promise_test(async () => {
+    const testTable = document.querySelector('#test-table');
+
+    const imageURL = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAMCAIAAAD3UuoiAAAAGklEQVQoz2Nk%2BP%2BfgRqAiYFKYNSgUYOGp0EA%2BQMCFrJdTgsAAAAASUVORK5CYII%3D";
+    for (const img of document.querySelectorAll('img.set-src')) {
+      img.src = imageURL;
+    }
+
+    const inputTypes = [
+      'text',
+      'search',
+      'tel',
+      'url',
+      'email',
+      'password',
+      'date',
+      'month',
+      'week',
+      'time',
+      'datetime-local',
+      'number',
+      'range',
+      'color',
+      'checkbox',
+      'radio',
+      'file',
+      'submit',
+      'image',
+      'image-with-src',
+      'reset',
+      'button',
+    ];
+    for (let inputType of inputTypes) {
+      let src = "data:,broken";
+      if (inputType == "image-with-src") {
+        inputType = "image";
+        src = imageURL;
+      }
+      let inputAtts;
+      if (inputType === 'image') {
+        inputAtts = `src="${src}" alt="x"`;
+      } else {
+        inputAtts = `value="x"`;
+      }
+      for (var appearanceValue of ["auto", "none"]) {
+        for (var overflowValue of ['visible', 'hidden', 'scroll']) {
+          const tr = document.createElement('tr');
+          tr.innerHTML = `<td><span>${inputType}</span> <input type=${inputType} ${inputAtts} style="overflow: ${overflowValue}; appearance: ${appearanceValue};"></td>`;
+          testTable.append(tr);
+        }
+      }
+    }
+
+    // wait for images to load
+    await new Promise(resolve => window.onload = e => resolve());
+    for (const img of document.images) {
+      assert_precondition(img.complete); // either error state or loaded
+    }
+
+    // get layout info from refs
+    const refTextOffsetTop = document.querySelector('.ref-text span').offsetTop;
+    const refCheckboxOffsetTop = document.querySelector('.ref-checkbox span').offsetTop;
+    const refColorOffsetTop = document.querySelector('.ref-color span').offsetTop;
+    const refFileOffsetTop = document.querySelector('.ref-file span').offsetTop;
+    const refImageAltOffsetTop = document.querySelector('.ref-image-alt span').offsetTop;
+    const refImageSrcOffsetTop = document.querySelector('.ref-image-src span').offsetTop;
+
+    function expectedOffsetTop(input) {
+      // TODO(zcorpan) https://github.com/whatwg/html/issues/5065
+      // for now this is intended to match Firefox
+      const style = input.getAttribute('style');
+      const src = input.getAttribute('src');
+      let rv;
+      switch (input.type) {
+        case 'file':
+          rv = refFileOffsetTop; break;
```

Comment:
Just returning from these would be a lot clearer and allow you to omit the break statement.